src/components/organisms/modals/EditableListModal/EditableListModal.tsx

```diff
@@ -0,0 +1,267 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import { FC, useCallback, useEffect, useMemo, useState } from 'react'
+import Button, {
+  AppearanceTypes,
+  IconPositioningTypes,
+  SizeTypes,
+} from 'components/molecules/Button/Button'
+import ModalBase, {
+  ButtonPositionTypes,
+  ModalSizeTypes,
+  TitleFontTypes,
+} from 'components/organisms/ModalBase/ModalBase'
+import { useTranslation } from 'react-i18next'
+import DynamicForm, {
+  FieldProps,
+  InputTypes,
+} from 'components/organisms/DynamicForm/DynamicForm'
+import {
+  filter,
+  includes,
+  isEqual,
+  join,
+  map,
+  omitBy,
+  reduce,
+  size,
+  split,
+  toNumber,
+} from 'lodash'
+import { FieldPath, SubmitHandler, useForm } from 'react-hook-form'
+import { ReactComponent as Add } from 'assets/icons/add.svg'
+import { TagTypes } from 'types/tags'
+import useAuth from 'hooks/useAuth'
+import classes from './classes.module.scss'
+import { Privileges } from 'types/privileges'
+import { ValidationError } from 'api/errorHandler'
+
+export enum DataStateTypes {
+  UPDATED = 'UPDATED',
+  NEW = 'NEW',
+  DELETED = 'DELETED',
+  OLD = 'OLD',
+}
+
+export type EditDataType = {
+  type?: TagTypes
+  name?: string
+  id?: string
+  state?: DataStateTypes
+}
+
+export interface EditableListModalProps {
+  isModalOpen?: boolean
+  closeModal: () => void
+  editableData?: EditDataType[]
+  title?: string
+  type?: TagTypes
+  isLoading?: boolean
+  handleOnSubmit?: (values: EditDataType[]) => void
```

Comment:
**[non-blocker]**

I think some of these could be required props as well. For example the handleOnSubmit function should probably be a required prop